src/service/location/location.service.ts

```diff
@@ -35,10 +34,59 @@ export class LocationService {
     }
   }
 
+  extractAddressFromString(multiAddr: string): Address {
+    return this.extractAddress(new Multiaddr(multiAddr));
+  }
+
+  extractAddressFromBase64(multiAddr: string): Address {
+    return this.extractAddress(new Multiaddr(Buffer.from(multiAddr, 'base64')));
+  }
+
+  async resolveAddress(address: Address): Promise<string> {
+    const cacheKey = this.getCacheKey(
+      this._resolveAddressCacheKey,
+      JSON.stringify(address),
+    );
+    const cachedData = await this.cacheManager.get<string>(cacheKey);
+    if (cachedData) return cachedData;
+
+    let result: string[];
+
+    switch (address.protocol) {
+      case 'dns4':
+        result = await resolve4(address.address);
+        break;
+      case 'dns6':
+        result = await resolve6(address.address);
+        break;
```

Comment:
Actually, now that I think of it, we shouldn't resolve the domain.
If the address is `/dns4/example.com/tcp/1234/http`, then we should hit `http://example.com:1234`, not `http://123.123.123.123:1234`. The difference is in contents of the `Host` HTTP header - if publisher is using any kind of virtual hosts, using IP directly may fail.

Reply:
This is probably as easy as just removing the `await resolve4` and `await resolve6` and leaving just `result = [address.address]`?

Reply:
this function is being used in 2 places:
1) for extracting the URL for ipni misreporting
2) for getting the location from ipinfo

for case 2) we still need an IP address instead of domain, but Ill change that for case 1)

Reply:
Ah true. Then indeed we need 2 different modes :/